lib/natalie/compiler/instructions/push_regexp_instruction.rb

```diff
@@ -12,7 +12,8 @@ def to_s
       end
 
       def generate(transform)
-        transform.exec_and_push(:regexp, "Value(RegexpObject::literal(env, #{@regexp.source.inspect}, #{@regexp.options}))")
+        source = @regexp.source.inspect.gsub(/\\#(\$|@)/, '#\1')
```

Comment:
I think there was a bug with Prism's `RegexpNode#unescaped` method so I didn't use it here. We should upgrade Prism and see if we can get this fix for free. I can look at that later today!

Reply:
That might work as well. We've got some test cases for it.

Reply:
Turns out it wasn't related to the Prism thing ... but I did remember we already have a `StringToCpp` module that handles this escaping for us, so I added a commit to your PR.

Reply:
Yeah, that looks so much better.